src/Symfony/Component/Validator/Constraints/NotBlankValidator.php

```diff
@@ -29,6 +29,10 @@ public function validate($value, Constraint $constraint)
             throw new UnexpectedTypeException($constraint, __NAMESPACE__.'\NotBlank');
         }
 
+        if (is_string($value) && null !== $constraint->normalizer) {
```

Comment:
I would move this after the `if` below. There's no need to apply the normalizer when the input is empty IMO.

Reply:
never mind, the result of the normalizer call could lead to an empty string too